### PR TITLE
Add pypi release github action

### LINKED ISSUE #604: Automate pypi release on gh tag?
Per maintainer meetup, here is a [workflow](https://github.com/nolanbconaway/shabadoo/blob/7fbc5bdfea5ff3e0d0357d788d5c3b09a1168dc7/.github/workflows/push_release_to_pypi.yml) i use to push a new releases to pypi whenever a GH tag is created. I'm never sure if i got the `on:` section just right but it seems to work for me!

```yaml
name: Release PyPi Version

on:
  release:
    types:
      - published

jobs:
  run:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v1
        with:
          python-version: "3.7"

      - name: Install Dependencies
        run: |
          pip install --upgrade pip
          pip install twine wheel --upgrade

      - name: Build Distribution
        run: python setup.py sdist bdist_wheel

      - name: Publish Python distribution to PyPI
        uses: pypa/gh-action-pypi-publish@master
        with:
          user: __token__
          password: ${{ secrets.PYPI_TOKEN }}
```

### PR DESCRIPTION
**Pull request by nolanbconaway**
Closes #604

This action will run whenever a new release is published on github, and will publish the version to pypi. I tested locally by running all the command in my venv, but this action also works well for some of my other projects.

The last remaining item is that @alanmcruickshank needs to set up `__token__` based authentication to PyPI and store the token password as a github secret called `PYPI_TOKEN`.

### CODE REVIEW

File: .github/publish-release-to-pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          pip install twine wheel --upgrade
+
+      - name: Build Distribution
+        run: python setup.py sdist bdist_wheel
+
+      - name: Publish Python distribution to PyPI
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
```
Comment (dmateusp) [inline .github/publish-release-to-pypi.yaml:29]: Missing a new line here ^^

**Review by dmateusp — COMMENTED** [review]
Would you mind adding documentation for the release process (letting future maintainers know that there is a workflow that runs on git tagging and the steps to tag and push a tag)

We should also note that the https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/config.ini file needs to be kept up to date

**Comment by dmateusp** [comment]
Once there is documentation and the token, @nolanbconaway do you want to test it by releasing `0.0.4a2` ?

**Comment by nolanbconaway** [comment]
That sounds good! I noticed that we hadn't been releasing alpha versions to pypi though-- is that something we'll do going forward?

**Comment by NiallRees** [comment]
> That sounds good! I noticed that we hadn't been releasing alpha versions to pypi though-- is that something we'll do going forward?

We did starting with 0.4.0a1 https://pypi.org/project/sqlfluff/0.4.0a1/

File: .github/publish-release-to-pypi.yaml (hunk 1 above)
Comment (nolanbconaway) [reply .github/publish-release-to-pypi.yaml]: nice catch!

**Comment by nolanbconaway** [comment]
Docs added! 
image

**Review by alanmcruickshank — APPROVED** [review]

**Comment by alanmcruickshank** [comment]
This looks great. I've added the PyPi token, and I've also sent @nolanbconaway an invite on pypi to be one of the maintainers. 

@nolanbconaway - how do you feel about merging this PR this weekend and then doing another dry run by releasing `0.4.0a2` as per @dmateusp comment?

**Comment by nolanbconaway** [comment]
> This looks great. I've added the PyPi token, and I've also sent @nolanbconaway an invite on pypi to be one of the maintainers.
> 
> @nolanbconaway - how do you feel about merging this PR this weekend and then doing another dry run by releasing `0.4.0a2` as per @dmateusp comment?

You betcha!

**Comment by nolanbconaway** [comment]
@sqlfluff/sqlfluff-maintainers LMK when 0.4.0a2 is 🎉  ready and i'll give the automated release a shot

**Pull request merged**